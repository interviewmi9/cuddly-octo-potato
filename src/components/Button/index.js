import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  align-items: center;
  background-color: ${props => props.theme.button.backgroundColor};
  border-radius: ${props => props.theme.borders.radius};
  border: ${props => props.theme.button.border};
  color: ${props => props.theme.button.color};
  cursor: pointer;
  display: inline-flex;
  flex-direction: row;
  font-size: ${props => props.theme.fontSizes.body};
  padding: 0.5rem 1rem;
  text-align: center;
  user-select: none;
  &:hover {
    background-color: ${props => props.theme.button.hover};
  }
`

const Button = ({ children, ...props }) => (
  <Wrapper onClick={() => props.onClick && props.onClick()} {...props}>{children}</Wrapper>
)

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
}

Button.defaultProps = {
  onClick: () => {},
}

export default Button
