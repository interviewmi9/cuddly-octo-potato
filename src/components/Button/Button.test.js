import React from 'react'
import { shallow } from 'enzyme'
import Button from './'

describe('button component', () => {
  it('renders correctly', () => {
    const rendered = render(<Button onClick={() => {}}>Test</Button>)
    expect(rendered).toMatchSnapshot()
  })

  it('renders a text message', () => {
    const text = 'Test'

    const button = shallow(<Button onClick={() => {}}>{text}</Button>)
    expect(button.contains(text)).toBeTruthy()
  })

  it('calls an action when clicked', () => {
    const fn = jest.fn()
    const btn = shallow(<Button onClick={fn}>Test</Button>)
    btn.simulate('click')
    expect(fn).toBeCalled()
  })
})
