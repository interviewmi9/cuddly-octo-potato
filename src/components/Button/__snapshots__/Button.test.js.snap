// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #376fc6;
  border-radius: 4px;
  border: solid 1px #376fc6;
  color: white;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 14px;
  padding: 0.5rem 1rem;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:hover {
  background-color: #3c92ef;
}

<div
  className="c0"
  onClick={[Function]}
>
  Test
</div>
`;
